src/spec/payid-protocol.md

```diff
@@ -259,38 +246,41 @@ This specification defines the PayID protocol - an application-layer protocol, w
+  The PayID client MAY specify the `Accept` request header field with at least one of the registered media types (Section X). The purpose of this header is to indicate what type of content can be understood in the response. It specifies the `paymentNetwork` and `environment` of the payment account and its representation format for which the PayID client wants to receive information. The representation format is always JSON.
 
-  The PayID server MUST reject formats that specify unknown or unsupported format parameters.
+  The PayID server MAY reject formats that specify unknown or unsupported format parameters.
 
 ## Response Headers
-  In addition to the Common Headers, the PayID server MUST specify the following response header.
+  In addition to the Common Headers, the PayID server SHOULD specify the following response header.
 
 ### Header Cache-Control
-  The PayID server MUST include the `Cache-Control` header with a value `Cache-Control: no-store`. PayID responses MUST NOT be stored in any cache.
+  The PayID server SHOULD include the `Cache-Control` header, to indicate how the PayID server operator wants PayID responses to be cached.
+
+###  Header Access-Control-Allow-Origin
+  The PayID server MUST include a `Access-Control-Allow-Origin: *` header, allowing all origins to make cross-origin requests to resolve PayIDs.
 
 # Extensibility
 
 ## Payload Extensibility
   PayID protocol supports extensibility in the payload, according to the specific format.
 
-  Regardless of the format, additional content MUST NOT be present if it needs to be understood by the receiver in order to correctly interpret the payload according to the specified PayID-Version header. Thus, clients MUST be prepared to handle or safely ignore any content not specifically defined in the version of the payload specified by the PayID-version header.
+  Regardless of the format, additional content MUST NOT be present if it needs to be understood by the receiver in order to correctly interpret the payload according to the specified PayID Version. Thus, clients MUST be prepared to handle or safely ignore any content not specifically defined in the version of the payload specified by the PayID Version.
 
 ## Header Field Extensibility
   The PayID protocol defines semantics around certain HTTP request and response headers. Services that support a version of the PayID protocol conform to the processing requirements for the headers defined by this specification for that version.
 
-  Individual services MAY define custom headers. These headers MUST NOT begin with PayID. Custom headers SHOULD be optional when making requests to the service. A service MUST NOT require the PayID client to understand custom headers to accurately interpret the response.
+  Individual services MUST NOT define custom headers, as custom headers complicate Cross-Origin Resource Sharing (CORS) requests.
```

Comment:
I changed this paragraph a bit, let me know if the reasoning makes sense.

Reply:
I think I prefer the original wording (or something closer to it) because I think we should guide implementers towards `SHOULD` behavior for broadest compatibility -- but if someone wants to define a custom header and make it mandatory, _and_ they don't care if certain clients break, then maybe that's fine for their use-case?

What about a middle ground between the original and new wording, like this:

```
Individual services SHOULD not define custom headers because custom headers complicate 
Cross-Origin Resource Sharing (CORS) requests and could impede broad interoperability. 
However, if an individual service defines custom headers, these headers MUST NOT begin
 with PayID, and SHOULD be optional when making requests to the service. A service 
MUST NOT require the PayID client to understand custom headers to accurately interpret 
the response.
```

Reply:
```
Individual services SHOULD NOT define custom headers because custom headers complicate 
Cross-Origin Resource Sharing (CORS) requests and could impede broad interoperability. 
However, if an individual service defines custom headers, these headers MUST NOT begin
 with PayID, and SHOULD be optional when making requests to the service. A service 
MUST NOT require the PayID client to understand custom headers to accurately interpret 
the response.
```

Could you explain more/give example on how custom headers complicate CORS?